### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by SanaTameem**
### In this milestone:
- I have added a new JavaScript file and imported it as a module(update status) which contains methods related to the status updates (completed: true/false).
- I have added an event listener to the checkbox.
- I have updated the items object's value for the completed key upon user actions.
- I have implemented a function for the "Clear all completed" button using the `filter()` method.
- The application is storing the updated array of items in local storage, so the user gets the correct list of values after the page reloads.

### For Web hint Error code reviewers (Abdelmejid Oumer Ali, Shahadat Hossain) and many more reviewers suggested us to submit with Error because the problem is universal and no one was able to Fix this problem.

error

### You can see that after running `npx hint . `it is showing zero Errors on my local machine but after pushing on GitHub it showing an Error.

### CODE REVIEW

File: modules/displayItem.js [hunk 1]
```diff
@@ -24,13 +23,23 @@ export const addTask = () => {
...
+// clearAll
+export const clearAll = () => {
+  arrayOfTasks = arrayOfTasks.filter((abc) => abc.completed === false);
+  arrayOfTasks = arrayOfTasks.map((task, index) => {
+    task.index = index + 1;
+    return task;
+  });
+  localStorage.setItem('arrayOfTasks', JSON.stringify(arrayOfTasks));
+  displayTasks();
+};
```
Comment (bobb-Rob) [inline modules/displayItem.js:41]: - [ ] Based on the project requirement, completed Items are cleared when the `clear all completed` button is clicked,  the task object is updated in the local storage as expected but the only challenge I see is when the page is refreshed, the items marked completed does not appear on the UI with a strikethrough as completed. Please see the image below 👇🏾 

image

Please ensure that when the page is refreshed the true state of the task object updates the UI correctly. If task `A` was marked as completed, when we refreshed, on the UI that task should have a strikethrough and the checkbox marked. .

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @SanaTameem, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/createItem.js [hunk 2]
```diff
@@ -38,9 +38,15 @@ export const createTaskElement = (todo) => {
...
-    localStorage.setItem('arrayOfTasks', JSON.stringify(arrayOfTasks));
+    statusUpdate(arrayOfTasks, inputText, checkbox, element, todo);
   });
+  // keeping the state of checkbox and input as it is:
+  if (todo.completed) {
+    inputText.classList.toggle('completed');
+    inputText.classList.toggle('line-through');
+    checkbox.checked = true;
+  }
+
```
Comment (Zuhaib042) [inline modules/createItem.js]: - [ ] The changes you have introduced have created a new bug in your app. right now the todos are not getting added to the list. I have tried multiple times but I could not add a todo. kindly consider fixing this bug.
Video Demonstration

https://github.com/SanaTameem/To-do-List/assets/96683843/d5657e39-0f75-406a-a203-1164943ee325

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @SanaTameem 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✅ Descriptive PR
✅ Professional Readme

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @SanaTameem , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  App works as expected
:heavy_check_mark: Great job working on changes requested

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**